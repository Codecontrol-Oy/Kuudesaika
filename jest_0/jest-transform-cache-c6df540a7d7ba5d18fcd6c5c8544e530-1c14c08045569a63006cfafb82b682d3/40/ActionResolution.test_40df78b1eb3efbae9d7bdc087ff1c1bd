'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _ActionResolution = require('./ActionResolution');var _ActionResolution2 = _interopRequireDefault(_ActionResolution);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('<ActionResolution />', function () {

    it('Should match snapshot', function () {
        var actionResolution = shallow(
        _react2.default.createElement(_ActionResolution2.default, { resolution: "" }));

        expect(actionResolution).toMatchSnapshot();
    });

    it('Should should render passed', function () {
        var actionResolution = mount(
        _react2.default.createElement(_ActionResolution2.default, { resolution: "PASSED" }));

        expect(actionResolution.find('h4').length).toEqual(1);
        expect(actionResolution.find('h4').first().text()).toEqual(' Hyväksytty ehdotuksen mukaan');

    });

    it('Should should render tabled', function () {
        var actionResolution = mount(
        _react2.default.createElement(_ActionResolution2.default, { resolution: "TABLED" }));

        expect(actionResolution.find('h4').length).toEqual(1);
        expect(actionResolution.find('h4').text()).toEqual(' Pöydätty');
    });
});