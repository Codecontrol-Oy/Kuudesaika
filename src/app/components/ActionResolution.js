import React from 'react';
import PropTypes from 'prop-types';
import CheckMarkIcon from 'grommet/components/icons/base/Checkmark';
import InformationIcon from 'grommet/components/icons/base/CircleInformation';
import ArchivedIcon from 'grommet/components/icons/base/Archive';
import RejectedIcon from 'grommet/components/icons/base/Close';
import ReturnedIcon from 'grommet/components/icons/base/Cycle';
import Heading from 'grommet/components/Heading';
import Section from 'grommet/components/Heading';
import theme from '../theme/global.scss';

export default class Breadcrumb extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
        action: null
    };
  }

  getResolution = (value) => {
        switch (value) {
            case 'PASSED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><CheckMarkIcon colorIndex={"ok"} /> Hyväksytty ehdotuksen mukaan</Heading>;
            }
            case 'PASSED_MODIFIED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><CheckMarkIcon colorIndex={"ok"} /> Hyväksytty esittelijän ehdotuksesta poiketen</Heading>;
            }
            case 'PASSED_REVISED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><CheckMarkIcon colorIndex={"ok"} /> Hyväksytty esittelijän muutetun ehdotuksen mukaan</Heading>;
            }
            case 'PASSED_VOTED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><CheckMarkIcon colorIndex={"ok"} /> Hyväksytty ehdotuksen mukaan äänestyksin</Heading>;
            }
            case '': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><InformationIcon colorIndex={"unknown"} /> Ei tietoa päätöksestä</Heading>;
            }
            case 'NOTED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><InformationIcon colorIndex={"unknown"} /> Merkittiin tiedoksi</Heading>;
            }
            case 'TABLED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><ArchivedIcon colorIndex={"unknown"} /> Pöydätty</Heading>;
            }
            case 'REJECTED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><RejectedIcon colorIndex={"critical"} /> Hylätty</Heading>;
            }
            case 'RETURNED': {
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><ReturnedIcon colorIndex={"warning"} /> Palautettu</Heading>;
            }
            default:
                return <Heading className={theme.marginTop} tag={"h4"} uppercase><ReturnedIcon colorIndex={"warning"} />-</Heading>;
        }
  }

  render () {
    return (
        <div>
            {this.getResolution(this.props.resolution)}
        </div>
    );
  }
}

Breadcrumb.propTypes = {
    resolution: PropTypes.string,
};