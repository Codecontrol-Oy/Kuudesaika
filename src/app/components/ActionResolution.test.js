import React from 'react';
import ActionResolution from './ActionResolution';

describe('<ActionResolution />', () => {

    it('Should match snapshot', () => {
        const actionResolution = shallow(
            <ActionResolution resolution={""} />
        );
        expect(actionResolution).toMatchSnapshot();
    });

    it('Should should render passed', () => {
        const actionResolution = mount(
            <ActionResolution resolution={"PASSED"} />
        );
        expect(actionResolution.find('h4').length).toEqual(1);
        expect(actionResolution.find('h4').first().text()).toEqual(' Hyväksytty ehdotuksen mukaan')

    });

    it('Should should render tabled', () => {
        const actionResolution = mount(
            <ActionResolution resolution={"TABLED"} />
        );
        expect(actionResolution.find('h4').length).toEqual(1);
        expect(actionResolution.find('h4').text()).toEqual(' Pöydätty')
    });
});