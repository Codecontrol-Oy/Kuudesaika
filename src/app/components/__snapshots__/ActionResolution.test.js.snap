// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionResolution /> Should match snapshot 1`] = `
<div>
  <Heading
    tag="h4"
    uppercase={true}
  >
    <CircleInformation
      colorIndex="unknown"
      responsive={true}
    />
     Ei tietoa päätöksestä
  </Heading>
</div>
`;
